manager/manager.go

```diff
@@ -676,24 +681,24 @@ func (self *manager) getRequestedContainers(containerName string, options v2.Req
 }
 
 func (self *manager) GetDirFsInfo(dir string) (v2.FsInfo, error) {
-	dirDevice, err := self.fsInfo.GetDirFsDevice(dir)
+	device, err := self.fsInfo.GetDirFsDevice(dir)
 	if err != nil {
-		return v2.FsInfo{}, fmt.Errorf("error trying to get filesystem Device for dir %v: err: %v", dir, err)
+		return v2.FsInfo{}, fmt.Errorf("failed to get device for dir %q: %v", dir, err)
 	}
-	dirMountpoint, err := self.fsInfo.GetMountpointForDevice(dirDevice.Device)
+	return self.getFsInfoByDeviceName(device.Device)
+}
+
+func (self *manager) GetFsInfoByFsUUID(uuid string) (*v2.FsInfo, error) {
+	device, err := self.fsInfo.GetDeviceInfoByFsUUID(uuid)
 	if err != nil {
-		return v2.FsInfo{}, fmt.Errorf("error trying to get MountPoint for Root Device: %v, err: %v", dirDevice, err)
+		return nil, fmt.Errorf("failed to get device using filesystem UUID %q: %v", uuid, err)
 	}
-	infos, err := self.GetFsInfo("")
-	if err != nil {
-		return v2.FsInfo{}, err
+	if device == nil {
+		// No such device exists.
+		return nil, nil
```

Comment:
I would prefer having an error type:
`ErrorNoSuchDevice = errors.New("No Such Device")`
Then we can check for that error in the kubelet if we want to.